### PR TITLE
GCS all types e2e tests

### PR DESCRIPTION
**Pull request by bharatgulati**

### CODE REVIEW

File: src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+   * obtained from Avro files converted to JSON.
+   *
+   * @param bucketName The name of the GCS bucket to validate.
+   * @return True if the GCS data matches the Avro data, false otherwise.
+   * @throws IOException If an IO error occurs during data retrieval.
+   */
+  public static boolean validateGCSSourceToGCSSink(String bucketName) throws IOException {
+    Map<String, JsonObject> expectedAvroData = convertAvroToJsonWithKeys();
+    Map<String, JsonObject> actualGcsData = listBucketObjects(bucketName);
+
```
Comment (yjhawar) [inline src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java]: nit: these lines in between aren't required

File: src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      .filter(blob -> blob.getSize() != 0)
+      .collect(Collectors.toList());
+
+    Stream<String> objectNamesWithData = bucketObjects.stream().map(blob -> blob.getName());
+    List<String> bucketObjectNames = objectNamesWithData.collect(Collectors.toList());
+
+    if (!bucketObjectNames.isEmpty()) {
+      String objectName = bucketObjectNames.get(0);
+      if (objectName.contains("part-r")) {
+        Map<String, JsonObject> dataMap2 = fetchObjectData(projectId, bucketName, objectName);
```
Comment (yjhawar) [inline src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java]: nit: suffix 2 isn't required here

File: src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+   * @throws IOException If an IO error occurs during Avro to JSON conversion.
+   */
+  public static Map<String, JsonObject> convertAvroToJsonWithKeys() throws IOException {
+    File avroFile = new File(avroFilePath);
+    DatumReader<GenericRecord> datumReader = new GenericDatumReader<>();
+    Map<String, JsonObject> avroDataMap = new HashMap<>();
+
+    try (FileReader<GenericRecord> dataFileReader = DataFileReader.openReader(avroFile, datumReader)) {
+      int keyCounter = 1;
+
```
Comment (yjhawar) [inline src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java]: nit: line not required

**Review by yjhawar — APPROVED** [review]
LGTM with minor comments

File: src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java (hunk 3 above)
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java]: Removed.

File: src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java (hunk 2 above)
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java]: Changed it to 'dataMap'

File: src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java (hunk 1 above)
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java]: Removed extra line.

**Comment by yjhawar** [comment]
Builds are failing. can you fix them

**Comment by bharatgulati** [comment]
> Builds are failing. can you fix them
 

Due to an intermittent problem (Chrome version issue), builds are failing. For the same, an issue was previously raised. [Link](https://cdap.atlassian.net/browse/PLUGIN-1649). An error message was changed but the cause is same.

**Pull request merged**